state_msg[] plant_states
string[] plant_paths
string node_name